File: Python Exams/Basic Exam Collection/Basics Online Retake Exam - 2 and 3 May 2019/04. Tourist Shop/04. Tourist Shop.py
budget = float(input())
spend_money = 0
buy_item_count = 0

item = input()

while item != 'Stop':
    buy_item_count += 1
    item_price = float(input())

    if buy_item_count % 3 == 0:
      spend_money += item_price / 2
    else:
      spend_money += item_price

    if budget < spend_money:
      print(f"You don't have enough money!\nYou need {spend_money - budget:.2f} leva!")
      exit()

    item = input()

print(f"You bought {buy_item_count} products for {spend_money:.2f} leva.")

File: JS Exams/01.  Basics Exam Collection/Basics Online Retake Exam - 2 and 3 May 2019/04. Tourist Shop/04. Tourist Shop.js
function touristShop(...arg) {
    let budget = +arg.shift()
    let spendMoney = 0
    let loop = 0

    for (let i = 1; i < arg.length; i += 2) {
      loop++
      loop % 3 === 0 ? spendMoney += +arg[i] / 2 :spendMoney += +arg[i]
    
      if (budget < spendMoney)
        return `You don't have enough money!\nYou need ${Math.abs(budget - spendMoney).toFixed(2)} leva!`

    } 

    return `You bought ${loop} products for ${spendMoney.toFixed(2)} leva.`
}

//console.log(touristShop(153.20, 'Backpack' , 25.20, 'Shoes', 54, 'Sunglasses', 30, 'Stop'))